lib/diagnostic.rb

```diff
@@ -7,36 +7,61 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    longest
+    list.each do |entry|
+      if longest.length < entry.length
+        longest = entry
+    end
+    longest
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    sum = 0
+    list.each do |entry|
+      sum += entry
+    end
+    sum
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    nubmers.each do |entry|
+      if (entry.odd?)
+        return true
+      end
+    end
+    return false
```

Comment:
Remember we don't need to use the `return` keyword in Ruby, the last line of the if statement or function will be returned for us. Here's how we can do this one:

`numbers.any?(&:odd?)`